middleware-jackson/src/main/java/com/nytimes/android/external/store3/middleware/jackson/JacksonParserFactory.java

```diff
@@ -56,10 +65,15 @@ private JacksonParserFactory() {
      * the provided {@link JsonFactory} instance.
      */
     @Nonnull
+    @SuppressWarnings("PMD.AvoidThrowingNullPointerException")
     public static <T> Parser<BufferedSource, T> createSourceParser(@Nonnull JsonFactory jsonFactory,
                                                                    @Nonnull Type type) {
-        Preconditions.checkNotNull(jsonFactory, "jsonFactory cannot be null.");
-        Preconditions.checkNotNull(type, "type cannot be null.");
+        if (jsonFactory == null) {
```

Comment:
Avoids an explicit dependency on cache that is only used for this (see PR description)